site/src/pages/WorkspacePage/WorkspaceReadyPage.tsx

```diff
@@ -242,19 +255,11 @@ export const WorkspaceReadyPage = ({
         }}
         handleCancel={cancelBuildMutation.mutate}
         handleSettings={() => navigate("settings")}
-        handleBuildRetry={() => {
-          switch (workspace.latest_build.transition) {
-            case "start":
-              startWorkspaceMutation.mutate({ logLevel: "debug" });
-              break;
-            case "stop":
-              stopWorkspaceMutation.mutate({ logLevel: "debug" });
-              break;
-            case "delete":
-              deleteWorkspaceMutation.mutate({ logLevel: "debug" });
-              break;
-          }
-        }}
+        handleBuildRetry={() => handleBuildRetry(false)}
+        handleBuildRetryDebug={() => handleBuildRetry(true)}
```

Comment:
Instead of passing 2 props here, can we just have `handleBuildRetry` and then pass it a boolean in the `Workspace` component?

Reply:
That would work – my main concern with setting up the functions the way I did was insulating the downstream components from changes, or having implementation details leak out

Right now, `handleBuildRetry` and `handleBuildRetryDebug` happen to use the same implementation under the hood, but that's just kind of by coincidence. If `WorkspaceReadyPage` ever gets to the point where the two functions need to look drastically different, we can change things in the one file, and none of the other components have to be any the wiser

I am aware that it does kind of increase the number of props (when we already have a lot), but keeping the other components blind to the implementation details seemed like the lesser of two evils to me

Reply:
> If WorkspaceReadyPage ever gets to the point where the two functions need to look drastically different, we can change things in the one file, and none of the other components have to be any the wiser
I think we might be pre-optimizing here, but I'm happy to keep as-is!